### PR TITLE
Added JSON-formatted output to CLI

### PR DESCRIPTION
**Pull request by juanfont**
This series of commits add support for a basic JSON output in all the Headscale commands.

The structure of the JSON is veeeery subject to changes, but this should help with the development of integration tests + automating the deployments of headscale.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,7 +148,15 @@ To get a certificate automatically via [Let's Encrypt](https://letsencrypt.org/)
 
 ## Disclaimer
 
-1. I have nothing to do with Tailscale, or Tailscale Inc. 
+1. We have nothing to do with Tailscale, or Tailscale Inc.
```
[the remark this answers is not in the document]
Comment (juanfont) [reply README.md]: Fixed!

File: cmd/headscale/headscale.go [hunk 2]
```diff
@@ -19,7 +20,27 @@ var versionCmd = &cobra.Command{
...
+
+		case "json-line":
+			j, err := json.Marshal(map[string]string{"version": version})
+			if err != nil {
+				log.Fatalln(err)
+			}
+			fmt.Println(string(j))
+		default:
+			fmt.Printf("Unknown format %s\n", o)
+		}
```
[the remark this answers is not in the document]
Comment (juanfont) [reply cmd/headscale/headscale.go]: Done :)

File: cmd/headscale/cli/nodes.go [hunk 3]
```diff
@@ -21,17 +22,22 @@ var RegisterCmd = &cobra.Command{
...
+			jsonOutput(m, err, o)
+			return
+		}
 		if err != nil {
-			fmt.Printf("Error: %s", err)
+			fmt.Printf("Cannot register machine: %s\n", err)
 			return
 		}
-		fmt.Println("Ook.")
+		fmt.Printf("Machine registered\n")
```
[the remark this answers is not in the document]
Comment (juanfont) [reply cmd/headscale/cli/nodes.go]: Yes, I fully agree with this. JsonOutput is a rather naive implementation to allow us moving forward towards a better testing suite.

But I have mixed feelings on adding subcommand-specific logic in the output function, e.g. including code there to deal with printing a list of routes from a node.

[the remark this answers is not in the document]
Comment (juanfont) [reply cmd/headscale/cli/nodes.go]: Should we merge the PR, and leave a issue open to track improvements on this?

**Comment by juanfont** [comment]
Should we merge the PR, and leave a issue open to track improvements on the human/JSON output?

File: cmd/headscale/cli/nodes.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (cure) [reply cmd/headscale/cli/nodes.go]: Yeah, I was thinking the subcommand output generation happens in each subcommand. That's why I'd pass both forms of output to the output function.

**Comment by cure** [comment]
> Should we merge the PR, and leave a issue open to track improvements on the human/JSON output?

Yeah I think that's fine! We can follow up with that refactoring later.

**Pull request merged**